File: src/main/java/com/wikia/tibia/objects/Street.java
package com.wikia.tibia.objects;

import com.wikia.tibia.utils.ObjectUtils;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class Street extends WikiObject {

    private City city;
    private City city2;
    private String map;
    private String floor;

    @Builder
    private Street(String name, String implemented, String notes, City city, City city2, String map, String floor) {
        super(name, null, null, null, implemented, notes, null, null);
        this.city = city;
        this.city2 = city2;
        this.map = map;
        this.floor = floor;
    }

    @Override
    public void setDefaultValues() {

        if (ObjectUtils.isEmpty(getImplemented())) {
            setImplemented("?");
        }
    }
}

File: src/main/kotlin/com/wikia/tibia/objects/Street.kt
package com.wikia.tibia.objects

import com.wikia.tibia.enums.City
import com.wikia.tibia.utils.ObjectUtils.isEmpty

data class Street(
        val name: String,
        val implemented: String,
        val notes: String,
        val city: City,
        val city2: City,
        val map: String,
        val floor: String
) : WikiObject() {

    override fun setDefaultValues() {
        if (isEmpty(implemented)) {
            this.copy(implemented = "?")
        }
    }
}